src/find_controls.cpp

```diff
@@ -105,14 +104,21 @@ FindEnforceAction GetFindEnforceActionImpl()
 
 FindEnforceAction GetFindEnforceAction()
 {
-    static const FindEnforceAction val = GetFindEnforceActionImpl();
+    static FindEnforceAction val = FindEnforceAction::Default_;
+    static std::string prev_env_str;
+    const auto& str = miopen::GetStringEnv(MIOPEN_FIND_ENFORCE{});
+    if(prev_env_str != str)
```

Comment:
Direct modifications of static variables are not MT-safe.

Reply:
Perhaps it would be overengineering to add a mutex here?
If I revert this to the previous state, all threads will perform the full env string parsing on each call to the function.
If I add a mutex around the logic in this function, every thread calling the function will have to wait to read out the value. But only one thread will actually do the work of parsing the string, and the string parsing would be summarily skipped in most cases.

Reply:
I would keep the legacy stuff in this PR and consider adding the ability to accept custom parsers to `struct EnvVar`.

Reply:
Reverting for now.

Reply:
[resolved]